File: Assets/Lua/Core/MsgID.lua
-- Assets/Lua/Core/MsgID.lua

-- 这个表的键名需与 C# MsgID 枚举成员同名，值与其在 C# 中的 ushort 值保持一致
local MsgID = {
    LoginRequest      = 1,
    LoginResponse     = 2,
}

-- 同时再导出一个列表，把名字映射给自己
MsgID.name = {
    [1] = "LoginRequest",
    [2] = "LoginResponse",
}
return MsgID

File: Assets/Scripts/Manager/MsgID.cs
namespace Game.Manager
{
	/// <summary>
	/// 所有消息 ID 枚举，客户端与服务器（或 Mock）需要保持一致。
	/// </summary>
	public enum MsgID : ushort
	{
		LoginRequest      = 1,
		LoginResponse     = 2,
		HeartbeatRequest  = 3,
		HeartbeatResponse = 4,
		// ───────────────────────────────────────────
		// 后续游戏业务消息统一往下追加，例如：
		// PlayerInfoUpdate = 5,
		// ChatMessage      = 6,
		// ……
	}
}